enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+- Version
+- InstallModes
+
+This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+
+Ex. _olm.yaml_
+
+```yaml
+name: packageserver-v1.14.1
+maturity: alpha
+version: 0.14.1
+replaces: packageserver-v1.14.0
+displayName: Package Server
+description: Represents an Operator package that is available from a given CatalogSource which will resolve to a ClusterServiceVersion.
+minKubeVersion: 1.11.0
+keywords: ['packagemanifests', 'olm', 'packages']
+maintainers:
+- name: Red Hat
+email: openshift-operators@redhat.com
+provider:
+name: Red Hat
+links:
+- name: Package Server
+url: https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/package-server
+installModes:
+- type: OwnNamespace
+  supported: true
+- type: SingleNamespace
+  supported: true
+- type: MultiNamespace
+  supported: true
+- type: AllNamespaces
+  supported: true
+icon: my-icon-png-base64-encoded
+labels:
+  operated-by: packageserver
+```
+
+#### `opm` Support
+
+[`opm`](https://github.com/operator-framework/operator-registry#overview) must be updated to provide support for building `k8s+v1` bundles. This can be acheived with explicit flags or through CSV/olm.yaml file presence.
+
+### CSV Synthesis
+
+OLM will support `k8s+v1` bundles by synthesizing an CSV from its contents. Resources usually generated by a CSV (e.g. APIService and Service) will not be directly applied, but instead be projected onto a synthesized CSV.
+
+This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
+
+### Indexes
+
+When a `k8s+v1` bundle is added to an index, `opm` will also synthesize and add a CSV. A lot of the existing `opm` and `operator-registry` code assumes bundles always contain CSVs, which make synthesis the simplest path to supporting `k8s+v1` bundles -- in the future, we may want to revisit this decision, especially when the [`Operator` API](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#proposal) is released.
+
+To fill out the dependency graph between versions of bundles, OLM considers the "replaces" field on the CSV which points to the previous version of the operator. This field will need to be supplied in the additional metadata the operator author provides in the bundle so that OLM can build dependency graphs. There is another enhancement around semantic versioning that would remove this requirement altogether, but it is independent of this enhancement.
+
+Supporting indexes with different mediatypes is a key requirement. Both `k8s+v1`  and `registry+v1` type bundles should be able to be added to an existing index. There is the potential of versioning the index itself to help differentiate between newer index images, that include the `k8s+v1` mediatype, and older versions.
```

Comment:
Since we are synthesizing CSV, what else could be different with respect to an index that would require to our version to differentiate k8s+v1 media type?

Reply:
The CSV needs to be synthesized in two places: index add and install

This is because we remove all content from the index for bundle references, save the update graph and the head CSV. At install, the cluster needs to know how to unpack the bundle content. If an older cluster that understands `registry+v1` and not `k8s+v1` tries to unpack the later, it will fail since it doesn't explicitly contain a CSV.

I think this means that we won't be able to have mixed indexes, or we need to be able to create `registry+v1` compatible indexes from mixed variants. We can always expand newer bundle references to "inline" their manifests, complete with synthesized CSV in a compatible catalog so that older clusters don't need to unpack them at runtime.